weblab/fitting/views.py

```diff
@@ -197,3 +204,150 @@ def get(self, request, *args, **kwargs):
         }
 
         return JsonResponse(response)
+
+
+class FittingResultCreateView(LoginRequiredMixin, PermissionRequiredMixin, UserFormKwargsMixin, FormView):
+    """
+    Create and submit a fitting result from models, protocols, fitting specs and datasets
+    and (where relevant) their versions.
+    """
+    permission_required = 'fitting.run_fits'
+    form_class = FittingResultCreateForm
+
+    template_name = 'fitting/fittingresult_create_form.html'
+
+    def get_initial(self):
+        initial = super().get_initial()
+        if 'model' in self.request.GET:
+            initial['model'] = get_object_or_404(ModelEntity, pk=self.request.GET['model'])
+        elif 'protocol' in self.request.GET:
+            initial['protocol'] = get_object_or_404(ProtocolEntity, pk=self.request.GET['protocol'])
+        elif 'fittingspec' in self.request.GET:
+            initial['fittingspec'] = get_object_or_404(FittingSpec, pk=self.request.GET['fittingspec'])
+        elif 'dataset' in self.request.GET:
+            initial['dataset'] = get_object_or_404(Dataset, pk=self.request.GET['dataset'])
+
+        return initial
+
+    def post(self, request, *args, **kwargs):
```

Comment:
Does the base class not provide this implementation as default? I think in other views we've done the logic in `post` rather than implementing `form_valid`; not sure what is considered Django best practice here?

Reply:
You're right - it's only when using `FormMixin` that this logic isn't provided. `FormView` does. I think it makes sense to follow the default Django logic in this case and just implement `form_valid` handling. I'll remove the `post` implementation.